products/kubic/main.pm

```diff
@@ -69,34 +68,23 @@ sub loadtests {
             loadtest 'shutdown/shutdown';
         }
     }
-
-    if ($filter eq 'feature') {
-        load_feature_tests;
-    }
-
-    if ($filter eq 'rcshell') {
-        # Tests before the YaST installation
-        loadtest 'caasp/rcshell_start';
-        loadtest 'caasp/libzypp_config';
-        loadtest 'caasp/one_line_checks';
-    }
 }
 
 #######################
 # Testing starts here #
 #######################
 if (get_var 'STACK_ROLE') {
-    loadtests 'boot_from_disk';
-    loadtests 'feature' if (check_var 'EXTRA', 'FEATURES');
+    load_boot_from_disk_tests;
+    load_feature_tests() if (check_var 'EXTRA', 'FEATURES');
     loadtest 'shutdown/shutdown';
 }
 else {
-    loadtests 'boot_from_dvd';
+    load_boot_from_dvd_tests;
     if (get_var 'SYSTEM_ROLE') {
         loadtests 'installation';
     }
     else {
-        loadtests('rcshell') if (check_var 'EXTRA', 'RCSHELL');
+        load_rcshell_tests() if (check_var 'EXTRA', 'RCSHELL');
```

Comment:
We could put it in the `elsif` to emphasize that we might not load any tests, I would recommend to add `die` for such a case. Also seems that this `main.pm` follows same design which SLE and openSUSE `main.pm` has, so be careful to redesign before we face same issues as there.